### PR TITLE
Improve the project API and documentation.

### PR DESCRIPTION
**Pull request by nsthorat**
Docs:
- Add a section for projects

API:
- Add `lilac init` and `lilac load`
- Add `set_project_dir` to set the environment variable globally.

Details:
- Deprecate `LILAC_DATA_PATH` in favor of `LILAC_PROJECT_DIR`.
- Add unit tests for load()

### CODE REVIEW

File: lilac/load.py [hunk 1]
```diff
@@ -3,64 +3,62 @@
...
+  """Load a project from a project configuration.
+
+  Args:
+    project_dir: The path to the project directory for where to create the dataset. If not defined,
+      uses the project directory from `LILAC_PROJECT_DIR` or [deprecated] `LILAC_DATA_PATH`. The
+      project_dir can be set globally with `set_project_dir`.
+    config: A Lilac config or the path to a json or yml file describing the configuration. The
+      contents should be an instance of `lilac.Config` or `lilac.DatasetConfig`. When not defined,
+      uses `LILAC_PROJECT_DIR`/lilac.yml.
+    overwrite: When True, runs all data from scratch, overwriting existing data. When false, only
```
Comment (dsmilkov) [inline lilac/load.py:44]: curious why we no longer pass `overwrite` to `_compute_embedding`?

**Review by dsmilkov — APPROVED** [review]
Amazing cleanup

File: docs/projects/projects.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+Lilac instance.
+
+An example project directory:
+
+```sh
+~/my_project
+    lilac.yml
+    datasets/
+        open-orca/
+    concepts/
```
[the remark this answers is not in the document]
Comment (nsthorat) [reply docs/projects/projects.md]: Renamed, we can deal with that later

File: docs/projects/projects.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+datasets: []
+```
+
+If you want to start a Lilac project without starting a webserver, you can use [](#init).
+
+From CLI:
+
+```sh
+lilac init ~/my_project
```
[the remark this answers is not in the document]
Comment (nsthorat) [reply docs/projects/projects.md:52]: Good question -- rill is basically UI-only (and configs). When using the python API, it's really weird that you have to start a server to start a project.

File: lilac/load.py (hunk 1 above)
Comment (nsthorat) [reply lilac/load.py:44]: We were doing the check for existence of signals/embeddings in 2 places, outside the worker, and inside. Now I just check overwrite outside the worker before scheduling a task. I have a unit test to make sure overwrite actually overwrites.

File: lilac/sources/gmail_source.py [hunk 4]
```diff
@@ -46,8 +49,8 @@ class GmailSource(Source):
   name = 'gmail'
 
   credentials_file: str = PydanticField(
-    description='Path to the OAuth credentials file.',
-    default=os.path.join(_GMAIL_CONFIG_DIR, _CREDS_FILENAME))
+    description=f'Path to the OAuth credentials file. Defaults to {_CREDS_FILENAME} in your Lilac '
```
[the remark this answers is not in the document]
Comment (nsthorat) [reply lilac/sources/gmail_source.py]: Done

File: lilac/sources/gmail_source.py [hunk 5]
```diff
@@ -46,8 +49,8 @@ class GmailSource(Source):
...
+    'project directory.')
```
[the remark this answers is not in the document]
Comment (nsthorat) [reply lilac/sources/gmail_source.py]: Done, I made this a `default_factory` so that it is fresh when environment variables change.

**Pull request merged**